x/emissions/migrations/v5/migrate_test.go

```diff
@@ -0,0 +1,388 @@
+package v4_test
+
+import (
+	"testing"
+
+	alloraMath "github.com/allora-network/allora-chain/math"
+
+	collections "cosmossdk.io/collections"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codecAddress "github.com/cosmos/cosmos-sdk/codec/address"
+
+	"cosmossdk.io/core/store"
+	"github.com/allora-network/allora-chain/app/params"
+
+	"cosmossdk.io/store/prefix"
+	"github.com/allora-network/allora-chain/x/emissions/keeper"
+	v5 "github.com/allora-network/allora-chain/x/emissions/migrations/v5"
+	emissions "github.com/allora-network/allora-chain/x/emissions/module"
+	emissionstestutil "github.com/allora-network/allora-chain/x/emissions/testutil"
+	emissionstypes "github.com/allora-network/allora-chain/x/emissions/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/suite"
+
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+
+	storetypes "cosmossdk.io/store/types"
+	cosmostestutil "github.com/cosmos/cosmos-sdk/testutil"
+)
+
+type EmissionsV4MigrationTestSuite struct {
+	suite.Suite
+	ctrl *gomock.Controller
+
+	ctx             sdk.Context
+	storeService    store.KVStoreService
+	emissionsKeeper *keeper.Keeper
+}
+
+func TestEmissionsV4MigrationTestSuite(t *testing.T) {
+	suite.Run(t, new(EmissionsV4MigrationTestSuite))
+}
+
+func (s *EmissionsV4MigrationTestSuite) SetupTest() {
+	encCfg := moduletestutil.MakeTestEncodingConfig(emissions.AppModule{})
+	key := storetypes.NewKVStoreKey(emissionstypes.StoreKey)
+	storeService := runtime.NewKVStoreService(key)
+	s.storeService = storeService
+	testCtx := cosmostestutil.DefaultContextWithDB(s.T(), key, storetypes.NewTransientStoreKey("transient_test"))
+	s.ctx = testCtx.Ctx
+
+	// gomock initializations
+	s.ctrl = gomock.NewController(s.T())
+	accountKeeper := emissionstestutil.NewMockAccountKeeper(s.ctrl)
+	bankKeeper := emissionstestutil.NewMockBankKeeper(s.ctrl)
+	emissionsKeeper := keeper.NewKeeper(
+		encCfg.Codec,
+		codecAddress.NewBech32Codec(params.Bech32PrefixAccAddr),
+		storeService,
+		accountKeeper,
+		bankKeeper,
+		authtypes.FeeCollectorName)
+
+	s.emissionsKeeper = &emissionsKeeper
+}
+
+// in this test, we check that an already migrated topic, that has all the new fields
+// for merit sortition, but has a NaN for initial regret, will have its initial regret
```

Comment:
I'm confused. Are we actually seeing NaN for initial regrets? I thought we were seeing very large values (huge enormous, negative values) which are non-NaN

Reply:
No NaNs afaik, just huge neg values

Reply:
so this comment can be updated then...